tree_walker.go

```diff
@@ -66,12 +66,16 @@ func (w *TreeWalker) Next() (name string, entry TreeEntry, obj Object, err error
 			return
 		}
 
-		obj, err = w.r.Object(entry.Hash)
-		if err == ErrObjectNotFound {
-			// FIXME: Avoid doing this here in case the caller actually cares about
-			//        missing objects.
+		if entry.Mode == 0160000 {
```

Comment:
Make this a constant (const submodule = "160000"), the left zero can be removed.